### PR TITLE
Fix some children types might be overwritten during transforming to path rules

### PR DESCRIPTION
**Pull request by moria97**
1. Fix some children types might be overwrited during transforming to path rules as shown in test cases.
2. Change FHIR path matching logic to a standard interface supported by [Hl7.FhirPath](https://github.com/FirelyTeam/fhir-net-common/blob/96b109975b55e065965866295efa1ddb80f39bb0/src/Hl7.FhirPath/FhirPath/TypedElementFPExtensions.cs).

### CODE REVIEW

File: src/Fhir.Anonymizer.Core/AnonymizationConfigurations/ResourceAnonymizerContext.cs [hunk 1]
```diff
@@ -44,18 +44,17 @@ private static void TransformTypeRulesToPathRules(ElementNode node, Dictionary<s
                 return;
             }
 
-            if (typeRules.ContainsKey(node.InstanceType))
+            if (!generatedTypePaths.Contains(path) && typeRules.ContainsKey(node.InstanceType))
```
Comment (BoyaWu10) [inline src/Fhir.Anonymizer.Core/AnonymizationConfigurations/ResourceAnonymizerContext.cs]: Shall we add some unit tests to explain these fix?

Comment (moria97) [reply src/Fhir.Anonymizer.Core/AnonymizationConfigurations/ResourceAnonymizerContext.cs]: Sure. Have added a case to address the problem.

File: src/Fhir.Anonymizer.Core.UnitTests/AnonymizerConfigurations/ResourceAnonymizerContextTests.cs [hunk 2]
```diff
@@ -29,6 +29,7 @@ public void GivenATypeRule_WhenParseRule_TransformedPathRuleShouldBeReturned()
 
             Assert.Contains("Patient.address", context.PathSet);
             Assert.Contains("Patient.name", context.PathSet);
+            Assert.Contains("Patient.name.period.start", context.PathSet);
```
Comment (moria97) [inline src/Fhir.Anonymizer.Core.UnitTests/AnonymizerConfigurations/ResourceAnonymizerContextTests.cs]: This test case will not pass in master code.

File: src/Fhir.Anonymizer.Core/AnonymizerEngine.cs [hunk 3]
```diff
@@ -74,9 +74,7 @@ public ElementNode AnonymizeResourceNode(ElementNode root)
             var resourceId = root.GetNodeId();
             foreach (var rule in resourceContext.RuleList)
             {
-                var pathCompileExpression = new FhirPathCompiler().Compile($"{rule.Path}");
-                var matchedNodes = pathCompileExpression(root, EvaluationContext.CreateDefault())
-                    .Cast<ElementNode>();
```
Comment (deepak-ms) [inline src/Fhir.Anonymizer.Core/AnonymizerEngine.cs]: Let me sync up with you on Teams to understand the details here.

**Pull request merged**